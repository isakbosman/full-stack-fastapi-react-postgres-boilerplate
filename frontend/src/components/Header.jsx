import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { appColor, headerHeight } from 'modules/theme';

import { logOut } from 'actions';

import { Container, utils } from 'styled-minimal';
import Icon from 'components/Icon';
import Logo from 'components/Logo';

const { responsive, spacer } = utils;

const HeaderWrapper = styled.header`
  background-color: #113740;
  height: ${headerHeight}px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 200;

  &:before {
    background-color: ${appColor};
    bottom: 0;
    content: '';
    height: 0.2rem;
    left: 0;
    position: absolute;
    right: 0;
  }
`;

const HeaderContainer = styled(Container)`
  align-items: center;
  display: flex;
  flex-wrap: wrap;
  height: 100%;
  justify-content: space-between;
  padding-bottom: ${spacer(2)};
  padding-top: ${spacer(2)};
`;

const Logout = styled.button`
  align-items: center;
  color: #fff;
  display: flex;
  font-size: 1.3rem;
  padding: ${spacer(2)};

  ${responsive({ lg: 'font-size: 1.6rem;' })}; /* stylelint-disable-line */

  &.active {
    color: #fff;
  }

  span {
    display: inline-block;
    margin-right: 0.4rem;
    text-transform: uppercase;
  }
`;

export default class Header extends React.PureComponent {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
  };

  handleClickLogout = () => {
    const { dispatch } = this.props;

    dispatch(logOut());
  };

  render() {
    return (
      <HeaderWrapper>
        <HeaderContainer>
          <Logo type="logo" />
          <Logout onClick={this.handleClickLogout}>
            <span>logout</span>
            <Icon name="sign-out" width={16} />
          </Logout>
        </HeaderContainer>
      </HeaderWrapper>
    );
  }
}
