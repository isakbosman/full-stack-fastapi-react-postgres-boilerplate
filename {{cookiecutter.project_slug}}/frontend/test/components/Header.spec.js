import React from 'react';

import Header from 'components/Header';

const mockDispatch = jest.fn();

function setup() {
  const props = {
    app: {},
    dispatch: mockDispatch,
    location: {
      pathname: '/',
    },
  };

  return mount(<Header suppressClassNameWarning {...props} />);
}

describe('Header', () => {
  const wrapper = setup();

  it('should render properly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should handle clicks', () => {
    wrapper.find('Logout').simulate('click');

    expect(mockDispatch).toHaveBeenCalledWith({
      type: 'USER_LOGOUT',
      payload: {},
    });
  });
});
