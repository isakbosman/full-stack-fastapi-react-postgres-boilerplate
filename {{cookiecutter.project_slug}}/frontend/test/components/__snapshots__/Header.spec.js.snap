// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render properly 1`] = `
.c2 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 0;
}

.c2 svg {
  height: 4.2rem;
  max-height: 100%;
  width: auto;
}

.c0 {
  background-color: #113740;
  height: 70px;
  left: 0;
  position: fixed;
  right: 0;
  top: 0;
  z-index: 200;
}

.c0:before {
  background-color: #00b4d5;
  bottom: 0;
  content: '';
  height: 0.2rem;
  left: 0;
  position: absolute;
  right: 0;
}

.c1 {
  box-sizing: border-box;
  margin-left: auto;
  margin-right: auto;
  padding-left: 16px;
  padding-right: 16px;
  max-width: 1440px;
  position: relative;
  width: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  height: 100%;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding-bottom: 8px;
  padding-top: 8px;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.3rem;
  padding: 8px;
}

.c3.active {
  color: #fff;
}

.c3 span {
  display: inline-block;
  margin-right: 0.4rem;
  text-transform: uppercase;
}

@media (min-width:768px) {
  .c1 {
    padding-left: 32px;
    padding-right: 32px;
  }
}

@media (min-width:1024px) {
  .c3 {
    font-size: 1.6rem;
  }
}

<HeaderWrapper>
  <header
    className="c0"
  >
    <HeaderContainer
      verticalPadding={false}
    >
      <div
        className="c1"
      >
        <Wrapper>
          <div
            className="c2"
          >
            SVG
          </div>
        </Wrapper>
        <Logout
          onClick={[Function]}
        >
          <button
            className="c3"
            onClick={[Function]}
          >
            <span>
              logout
            </span>
            <IconWrapper
              src="/media/icons/sign-out.svg"
              width={16}
            />
          </button>
        </Logout>
      </div>
    </HeaderContainer>
  </header>
</HeaderWrapper>
`;
